source/Server/IntegratedAuthentication/IntegratedAuthenticationHost.cs

```diff
@@ -0,0 +1,135 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.HttpSys;
+using Octopus.Diagnostics;
+using Octopus.Server.Extensibility.Authentication.DirectoryServices.Configuration;
+using Octopus.Server.Extensibility.Authentication.DirectoryServices.DirectoryServices;
+using Octopus.Server.Extensibility.Extensions.Infrastructure.Web;
+using Octopus.Server.Extensibility.HostServices.Web;
+
+namespace Octopus.Server.Extensibility.Authentication.DirectoryServices.IntegratedAuthentication
+{
+    class IntegratedAuthenticationHost : IExecuteWhenServerStarts
+    {
+        readonly ILog log;
+        readonly IWebPortalConfigurationStore configuration;
+        readonly IDirectoryServicesConfigurationStore configurationStore;
+        readonly IIntegratedAuthenticationHandler handler;
+
+        public IntegratedAuthenticationHost(ILog log,
+            IWebPortalConfigurationStore configuration,
+            IDirectoryServicesConfigurationStore configurationStore,
+            IIntegratedAuthenticationHandler handler)
+        {
+            this.log = log;
+            this.configuration = configuration;
+            this.configurationStore = configurationStore;
+            this.handler = handler;
+        }
+
+        public void OnHostStarting() {}
+
+        public void OnHostStarted()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
I thought we were also going to check if the Ad integration was enabled or disabled?

Reply:
So should we also do `directoryServicesConfigurationStore.GetIsEnabled()` here?